src/test_l1_tvSettings.c

```diff
@@ -809,8 +809,10 @@ void test_l1_tvSettings_positive_GetTVSupportedVideoFormats (void)
 
     tvError_t result = tvERROR_NONE;
     char keyValue[UT_KVP_MAX_ELEMENT_SIZE] = { 0 };
-    tvVideoFormatType_t *tvVideoFormats[1]={0};
-    tvVideoFormatType_t *tvVideoFormatsRetry[1]={0};
+    tvVideoFormatType_t tvVideoFormats[VIDEO_FORMAT_MAX];
+    tvVideoFormatType_t *tvVideoFormatsPtr = tvVideoFormats;
```

Comment:
These three functions,

GetTVSupportedDolbyVisionModes
GetTVSupportedVideoFormats
GetTVSupportedDimmingModes
are passing an array of pointers as the first parameter. However, we noticed that these APIs are not correctly utilizing the arrays of pointers in the amologic code. We can modify the L1 / L2 code to match the amologic code.

Reply:
As discussed, Please modify the code handling to match the array of pointers straight away.

Reply:
As per our discussed, you were suggesting to implement it this way considering the future integration of mediatek code.
    tvVideoFormatType_t *videoFormats[MAX_VIDEO_FORMATS];
    unsigned short count = 0;
 
    // Call the function to get supported video formats
    if (GetTVSupportedVideoFormats(videoFormats, &count) == TV_ERROR_NONE) {
    ""
    ""
    }
    
    Note:
    This creates an array of pointers, but the pointers themselves are uninitialized. They do not point to valid memory.
    
    
    I hope it is fine to proceed this way.

Reply:
This declaration creates an array of pointers to tvVideoFormatType_t. The array itself is valid, 

The implementation is valid if GetTVSupportedVideoFormats assigns valid memory addresses to the pointers in videoFormats and updates count correctly. The array itself does not need prior initialization, as the function is responsible for assigning the pointers.

Reply:
Okay. I have updated as per your feedback.